### ISSUE TITLE
there is no log and processbar when running trainer.train()

### ISSUE THREAD

**Issue by yezhipeng2417** [huggingface/transformers#19884]
### System Info

When I open a .ipynb in vscode or datashell, there is no log and processbar when running trainer.train(), but in jupyter notebook, it shows.
<img width="770" alt="截屏2022-10-26 11 55 46" src="https://user-images.githubusercontent.com/87161948/197910485-5d16a652-48a4-4c83-b110-b9ad7a95046b.png">

<img width="1138" alt="截屏2022-10-26 12 06 12" src="https://user-images.githubusercontent.com/87161948/197910740-3f7ba219-f27c-4a76-82ac-827109d9257e.png">

### Who can help?

@sgugger 

### Information

- [ ] The official example scripts
- [X] My own modified scripts

### Tasks

- [X] An officially supported task in the `examples` folder (such as GLUE/SQuAD, ...)
- [ ] My own task or dataset (give details below)

### Reproduction

`from transformers import AutoModelForSequenceClassification

model = AutoModelForSequenceClassification.from_pretrained('bert-base-cased',
                                                           num_labels=2)

print(sum([i.nelement() for i in model.parameters()]) / 10000)
import numpy as np
from datasets import load_metric
from transformers.trainer_utils import EvalPrediction
metric = load_metric('accuracy')

def compute_metrics(eval_pred):
    logits, labels = eval_pred
    logits = logits.argmax(axis=1)
    return metric.compute(predictions=logits, references=labels)

eval_pred = EvalPrediction(
    predictions=np.array([[0, 1], [2, 3], [4, 5], [6, 7]]),
    label_ids=np.array([1, 1, 1, 1]),
)

compute_metrics(eval_pred)
from transformers import TrainingArguments, Trainer

args = TrainingArguments(output_dir='./output_dir', evaluation_strategy='epoch')
args.num_train_epochs = 1
args.learning_rate = 1e-4
args.weight_decay = 1e-2
args.per_device_eval_batch_size = 32
args.per_device_train_batch_size = 16

trainer = Trainer(
    model=model,
    args=args,
    train_dataset=dataset_train,
    eval_dataset=dataset_test,
    compute_metrics=compute_metrics,
)
# train
trainer.train()`

### Expected behavior

I hope it will output totally the same in all of the three IDEs

**Comment by sgugger** [comment]
Vscode or datashell do not support widgets, as far as I know, so we can't show the same progress bars there as in a notebook.

**Comment by yezhipeng2417** [comment]
thanks
